File: src/python/src/de/yaxgl/Controls/Button.py
import wx
from de.yaxgl.Base.Control import Control



class Button(Control):
    
    def __init__(self, owner, ID):
        Control.__init__(self)
        self.owner = owner
        self.ID = ID
        self.control = wx.Button(parent=owner.getNativeComponent())

        self.control.Bind(wx.EVT_BUTTON, self.onButtonClick)

    def initializeNativeControl(self, xmlElement):
        self.control.SetDimensions(int(xmlElement.getAttribute("xpos")),
                                   int(xmlElement.getAttribute("ypos")),
                                   int(xmlElement.getAttribute("width")),
                                   int(xmlElement.getAttribute("height"))
                                   )
        self.setLabel(xmlElement.getAttribute("label"))
    
    def getLabel(self):
        return(self.control.GetLabelText())
    
    def setLabel(self, label):
        self.control.SetLabel(label)

    def onButtonClick(self, event):
        self.clickEvent(self, event)

File: src/java/YaxglLibrary/de/yaxgl/Controls/Button.java
/**
 * 
 */
package de.yaxgl.Controls;

import org.w3c.dom.Element;

import org.eclipse.swt.*;
import de.yaxgl.Base.Control;
import de.yaxgl.Container.Container;


public class Button extends Control {

	public Button(Container owner,String ID)
	{
		this.ID=ID;
		this.owner=owner;
		this.control=new org.eclipse.swt.widgets.Button((org.eclipse.swt.widgets.Composite)owner.getNativeComponent(),SWT.PUSH);
		/*register Events*/
		((org.eclipse.swt.widgets.Button)this.control).addSelectionListener(clickEvent(this));
		this.control.addFocusListener(focusEvent(this));
		
	}
	
	@Override
	public void initializeNativeControl(Element xmlElement) {
		
		
		
		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));

        setLabel(xmlElement.getAttribute("label"));
		
	}
	
	 public String getLabel()
     {
         return ((org.eclipse.swt.widgets.Button)this.control).getText();
     }

     public void setLabel(String label)
     {
    	 ((org.eclipse.swt.widgets.Button)this.control).setText(label);
     }

}
